http-gateway/service/getResource.go

```diff
@@ -1,29 +1,63 @@
 package service
 
 import (
+	"encoding/base64"
 	"net/http"
 	"net/http/httptest"
 	"strings"
 
 	"github.com/google/go-querystring/query"
 	"github.com/gorilla/mux"
+	"github.com/plgd-dev/hub/v2/grpc-gateway/pb"
 	"github.com/plgd-dev/hub/v2/http-gateway/serverMux"
 	"github.com/plgd-dev/hub/v2/http-gateway/uri"
 	kitNetGrpc "github.com/plgd-dev/hub/v2/pkg/net/grpc"
 	"github.com/plgd-dev/hub/v2/resource-aggregate/commands"
 	"google.golang.org/grpc/codes"
 )
 
-func (requestHandler *RequestHandler) getResourceFromTwin(w http.ResponseWriter, r *http.Request, resourceID string) {
+const errFmtFromTwin = "cannot get resource('%v') from twin: %w"
+
+func getHeaderETag(r *http.Request) string {
+	etagStr := r.Header.Get(uri.ETagHeaderKey)
+	if etagStr != "" {
+		return etagStr
+	}
+	return r.Header.Get(strings.ToLower(uri.ETagHeaderKey))
+}
+
+func getETags(r *http.Request) [][]byte {
+	etagStr := getHeaderETag(r)
+	if etagStr != "" {
+		if etag, err := base64.StdEncoding.DecodeString(etagStr); err == nil {
+			return [][]byte{etag}
+		}
+	}
+	etagQ := r.URL.Query()[uri.ETagQueryKey]
+	if len(etagQ) == 0 {
+		return nil
+	}
+	etags := make([][]byte, 0, len(etagStr))
```

Comment:
`etagStr` is equal to `""` in this code branch (see line 31)

Reply:
You are right :)